harness/determined/common/api/authentication.py

```diff
@@ -145,24 +142,19 @@ def do_login(
     password: str,
     cert: Optional[certs.Cert] = None,
 ) -> str:
-    r = api.post(
-        master_address,
-        "login",
-        json={"username": username, "password": password},
-        authenticated=False,
-        cert=cert,
-    )
-
-    token = r.json()["token"]
-    assert isinstance(token, str), "got invalid token response from server"
+    password = api.salt_and_hash(password)
+    unauth_session = api.Session(user=username, master=master_address, auth=None, cert=cert)
+    login = bindings.v1LoginRequest(username=username, password=password, isHashed=True)
+    r = bindings.post_Login(session=unauth_session, body=login)
+    token = r.token
 
     return token
 
 
 def _is_token_valid(master_address: str, token: str, cert: Optional[certs.Cert]) -> bool:
     """
     Find out whether the given token is valid by attempting to use it
-    on the "/users/me" endpoint.
+    on the "api/v1/me" endpoint.
     """
     headers = {"Authorization": "Bearer {}".format(token)}
     try:
```

Comment:
if I use bindings.get_GetMe here I won't be able to use the token. Is that ok?

Reply:
yeah, I can't think of another way. you also need to update the actual request URL as well though?